File: Week Assignment 4 using C/5_josephus_problem.c
// 5. Josephus Problem (GFG classic)
#include <stdio.h>
#include <stdlib.h>

// Recursive solution for Josephus (1-based indexing)
int josephus(int n, int k) {
    if (n == 1) return 1;
    return (josephus(n-1, k) + k - 1) % n + 1;
}

int main() {
    printf("%d\n", josephus(7, 3)); // expected 4 (for n=7,k=3)
    printf("%d\n", josephus(5, 2)); // expected 3
    return 0;
}

File: Week Assignement 4 using Python/5_josephus_problem.py
# 5. Josephus problem (recursive)
def josephus(n,k):
    if n==1: return 1
    return (josephus(n-1,k)+k-1) % n + 1

if __name__ == "__main__":
    print(josephus(7,3))
    print(josephus(5,2))
